File: AUGUST/shlee/BOJ2167.py
N,M = map(int,input().split())
arr = []
for i in range(N):
    row = list(map(int,input().split()))
    arr.append(row)

K = int(input())
for _ in range(K):
    i,j,x,y = map(int,input().split())
    Sum = 0
    for r in range(i,x+1):
        for c in range(j,y+1):
            Sum += arr[r-1][c-1]

    print(Sum)

File: AUGUST/KINHYEONJI/BOJ2167.cpp
#include <iostream>

using namespace std;

int main() {
    int N, M;
    cin >> N >> M;
    int arr[N][M];
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < M; j++) {
            cin >> arr[i][j];
        }
    }
    int count;
    cin >> count;
    while (count--) {
        int i, j, x, y;
        cin >> i >> j >> x >> y;
        int sum = 0;
        for (int row = i - 1; row < x; row++) {
            for (int col = j - 1; col < y; col++) {
                sum += arr[row][col];
            }
        }

        cout << sum << '\n';
    }
}